salt/_modules/metalk8s_etcd.py

```diff
@@ -0,0 +1,98 @@
+# -*- coding: utf-8 -*-
+'''
+Module for handling etcd client specific calls.
+'''
+import logging
+
+from salt.exceptions import CommandExecutionError
+from salt.utils.decorators import depends
+
+PYTHON_ETCD_PRESENT = False
+try:
+    import etcd3
+    PYTHON_ETCD_PRESENT = True
+except ImportError:
+    pass
+
+
+log = logging.getLogger(__name__)
+
+__virtualname__ = 'metalk8s_etcd'
+
+
+def __virtual__():
+    return __virtualname__
+
+
+def deps_missing(*args, **kwargs):
+    raise CommandExecutionError("python etcd client is not installed")
+
+
+def _get_endpoint_up(ca_cert, cert_key, cert_cert):
+    """Find an etcd endpoint among all servers declared in mine"""
+    endpoints = __salt__['saltutil.runner'](
```

Comment:
No. We know which servers are part (or are supposed to be part of) the `etcd` ensemble. No need to go probe every single minion or whatnot.

Reply:
I'm not sure to understand this one, how do we know which servers are part of the etcd cluster? I did not find anything special in grains. Or do you mean query the kubeapi with `python-kubernetes` and find out which node have the etcd roles?

Reply:
There's an existing macro to derive this from the pillar.

Reply:
will fix